test/signhash.ts

```diff
@@ -115,6 +116,49 @@ contract('SignHash', accounts => {
     });
   });
 
+  describe('#getSigners', () => {
+    it('should return empty list when there is no signers', async () => {
+      const signers = await instance.getSigners(hash, maxCount);
+      assert.deepEqual(signers, []);
+    });
+
+    it('should return single element list for a single signer', async () => {
+      await instance.sign(hash);
+
+      const signers = await instance.getSigners(hash, maxCount);
+      assert.deepEqual(signers, [defaultAccount]);
+    });
+
+    it('should return signers in chronological order', async () => {
+      for (const account of accounts) {
+        await instance.sign(hash, { from: account });
+      }
+
+      const signers = await instance.getSigners(hash, maxCount);
+      assert.deepEqual(signers, accounts);
+    });
+
+    it('should trim signers to maxCount', async () => {
+      for (const account of accounts) {
+        await instance.sign(hash, { from: account });
+      }
+
+      const reducedCount = maxCount - 1;
+      const signers = await instance.getSigners(hash, reducedCount);
+      assert.deepEqual(signers, accounts.slice(0, reducedCount));
+    });
+
+    it('should return empty list when maxCount is zero', async () => {
+      for (const account of accounts) {
+        await instance.sign(hash, { from: account });
+      }
```

Comment:
This `for` is repeated so many times it might be worth extracting it to a function

Reply:
You are right. I tried to replace it with a functional style one-liner, but I could not come up with something clean and at the same time maintaining the order of accounts. I extracted the loop to a helper function. Much cleaner now.